### PR TITLE
fix(Table): style value should support CSS var

### PR DESCRIPTION
**Pull request by li-jia-nan**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [x] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | fix(Table): style value should support CSS var |
| 🇨🇳 Chinese | fix(Table): style value should support CSS var |

### CODE REVIEW

**Review by afc163 — APPROVED** [review]

**Comment by li-jia-nan** [comment]
@afc163 大佬手好快，其实我不确定这个  Visual diff 是不是预期的

**Comment by afc163** [comment]
图片

确实不太对。

File: components/table/style/selection.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ const genSelectionStyle: GenerateStyle<TableToken, CSSObject> = (token) => {
       },
 
       [`table tr th${componentCls}-selection-column${componentCls}-cell-fix-left`]: {
-        zIndex: token.zIndexTableFixed + 1,
+        zIndex: calc(token.zIndexTableFixed).add(1).equal(),
```
Comment (afc163) [inline components/table/style/selection.ts:64]: 图片

会算成 px 值。

Comment (li-jia-nan) [reply components/table/style/selection.ts:64]: - #50357

**Comment by afc163** [comment]
这个问题有重现么？

**Comment by li-jia-nan** [comment]
> 这个问题有重现么？

没有，这里其实不算 bug，只是按照 css var 的规范，所有的计算应该交给 css calc 去做，而不是放在 js 运行时

**Comment by li-jia-nan** [comment]
> 这个问题有重现么？

比如之前的结果是 3，现在变成了 calc(2 +1)，更符合 css var 的规范

**Pull request merged**